riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyRequestSizeValidationComponentTest.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    @Test
+    public void should_return_bad_request_when_request_exceeds_endpoint_overridden_configured_max_request_size() throws IOException {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpointWithRequestSizeValidationOverride.MATCHING_PATH)
+                        .log().all()
+                        .body(generatePayloadOfSizeInBytes(BasicEndpointWithRequestSizeValidationOverride.MAX_REQUEST_SIZE + 1))
+                        .when()
+                        .post()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.BAD_REQUEST.code());
+        assertBadRequestErrorMessageAndMetadata(response);
+    }
+
+    @Test
+    public void should_return_expected_response_when_not_exceeding_endpoint_overridden_request_size() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpointWithRequestSizeValidationOverride.MATCHING_PATH)
+                        .log().all()
+                        .body(generatePayloadOfSizeInBytes(BasicEndpointWithRequestSizeValidationOverride.MAX_REQUEST_SIZE))
+                        .when()
+                        .post()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+        assertThat(response.asString()).isEqualTo(BasicEndpointWithRequestSizeValidationOverride.RESPONSE_PAYLOAD);
+    }
+
+    @Test
+    public void should_return_expected_response_when_endpoint_disabled_request_size_validation() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpointWithRequestSizeValidationDisabled.MATCHING_PATH)
+                        .log().all()
+                        .body(generatePayloadOfSizeInBytes(GLOBAL_MAX_REQUEST_SIZE + 100))
+                        .when()
+                        .post()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+        assertThat(response.asString()).isEqualTo(BasicEndpointWithRequestSizeValidationDisabled.RESPONSE_PAYLOAD);
+    }
+
+    private static String generatePayloadOfSizeInBytes(int length) {
+        StringBuilder sb = new StringBuilder(length);
+        for(int i = 0; i < length; i++) {
+            sb.append(i);
```

Comment:
For anything i >= 10 you'll be getting a string larger than the requested length in bytes. You could do `sb.append(i % 10)`.